pandas/_libs/tslibs/timedeltas.pyi

```diff
@@ -67,10 +67,19 @@ class Timedelta(timedelta):
     def __abs__(self) -> timedelta: ...
     def __mul__(self, other: float) -> timedelta: ...
     def __rmul__(self, other: float) -> timedelta: ...
-    @overload
+    # error: Signature of "__floordiv__" incompatible with supertype "timedelta"
+    @overload  # type: ignore[override]
     def __floordiv__(self, other: timedelta) -> int: ...
     @overload
-    def __floordiv__(self, other: int) -> timedelta: ...
+    def __floordiv__(self, other: int | float) -> timedelta: ...
+    @overload
+    def __floordiv__(self, other: np.ndarray) -> npt.NDArray[np.timedelta64]: ...
```

Comment:
doesnt the result dtype depend on the dtype for `other`?  e.g. if other is td64 then the result would be int64

Reply:
I think it is always an array of timedelta64: https://github.com/pandas-dev/pandas/blob/9cc98a064e2873ff5ddab62efdb23060c2d52167/pandas/_libs/tslibs/timedeltas.pyx#L1552 but it could also just be a single Timedelta.

Reply:
The following line has checks for other.dtype.kind with different behavior for "m" vs ["i", "u", "f"]

Reply:
If `other` is `npt.NDArray[np.timedelta64]`, it returns `npt.NDArray[np.intp]`, in the other cases it returns a `npt.NDArray[np.timedelta64]`.

I added overload for this. Since, `np.timedelta64` is not a sub-class of `np.number` https://numpy.org/devdocs/reference/typing.html#timedelta64, this makes the overloads a bit easier :)